akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -51,6 +54,18 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
    */
   def lastStreamId(): Int = largestIncomingStreamId
 
+  private var maxConcurrentStreams = Http2Protocol.InitialMaxConcurrentStreams
+  def setMaxConcurrentStreams(newValue: Int): Unit = maxConcurrentStreams = newValue
+  /**
+   * @return true is the number of outgoing streams in Open (and HalfClosedXxx)
+   *         state doesn't exceed MaxConcurrentStreams
+   */
+  def hasOutgoingCapacity: Boolean = {
+    // StreamStates only contains streams in active states (active states are any variation
+    // of Open, HalfClosed) so using the `size` works fine to compute the capacity
+    streamStates.size < maxConcurrentStreams
+  }
```

Comment:
Once I realized that I should only generate demand _after_ `streamStates` had been updated, I could get rid of an ugly `-1` required here.